treeherder/etl/buildapi.py

```diff
@@ -307,6 +334,13 @@ def transform(self, data, source, filter_to_revision=None, filter_to_project=Non
                 # using project and revision form the revision lookups
                 # to filter those jobs with unmatched revision
                 for job in jobs:
+                    job_ids_seen_now.add(job['id'])
```

Comment:
This is not an issue, just a comment: I suspect you get better performance if you create the job_ids_seen_now set in one shot out of this loop, ie
```python
job_ids_seen_now = set([job['id'] for job in jobs])
for job in jobs:
```
 If you do that, you can also iterate directly on the difference between job_ids_seen_now and job_ids_seen_last_time which makes it more readable imho :smiley_cat: . As I said, this is just a comment, feel free to drop it.